follower/follower_test.go

```diff
@@ -91,6 +92,14 @@ func TestMain(m *testing.M) {
 	tmpDir, _ = ioutil.TempDir("", "fllw")
 	rs := m.Run()
 	os.RemoveAll(tmpDir)
+	if rs == 0 {
+		// Followers may take 10 seconds to notice the removal.
+		time.Sleep(10 * time.Second)
+		if runtime.NumGoroutine() > 2 {
+			// Heuristic to detect leaked goroutines.
+			rs = 1
+		}
+	}
```

Comment:
Are you able to put this in a separate test case that can communicate why the test failed?